lib/ansible/utils/display.py

```diff
@@ -48,6 +49,50 @@
     # Python 3, we already have raw_input
     pass
 
+# Set the locale to the users default setting
+locale.setlocale(locale.LC_ALL, '')
```

Comment:
It feels wrong to put this in display because it's a program global initialization (which can have an effect on other things than just wcwidth, for instance).  Perhaps we need a layer of abstraction for things that are needed to initialize the environment which ansible runs in?

``` python
def first_initialization():
    global first_initialization_finished
    locale.setlocale(locale.LC_ALL, '')
   # Could also do things like set up an initial logger [prior to reading configuration of logging] here
    first_initialization_finished = True
# Call first_initialization() from the cli code somewhere
```

and then document in display.get_text_width() that the initialization function has to be run prior to running this?  Perhaps using a global variable to enforce and throw a nice error if it isn't?

``` python
def get_text_width(text):
    if not ansible.initialiation_module.first_initialization_finished:
        display.error('danger will robinson')
[...]
```

Reply:
I originally had it in the CLI, but since this code is dependent on it, it felt wrong to have code that depends on it, not be responsible for ensuring it was set.  It made it more complex to use, for example in the case of tests, the tests had to be responsible for setting it.

I'm still on the fence here.